sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/RocksDB.scala

```diff
@@ -226,6 +226,86 @@ class RocksDB(
     this
   }
 
+  /**
+   * Load from the start checkpoint version and apply all the changelog records to reach the
+   * end version. Note that this will copy all the necessary file from DFS to local disk as needed,
+   * and possibly restart the native RocksDB instance.
+   *
+   * @param startVersion
+   * @param endVersion
+   * @param readOnly
+   * @return
+   */
+  def load(startVersion: Long, endVersion: Long, readOnly: Boolean): RocksDB = {
+    assert(startVersion >= 0 && endVersion >= startVersion)
+    acquire(LoadStore)
```

Comment:
lock release path is still the same right ? i assume we release on an abort ?

Reply:
I think I am copying the existing implementation. Any changes needed here?

Reply:
Yea - Im guessing the unlock happens in the end as part of an abort within the state data source reader